lib/currency.js

```diff
@@ -0,0 +1,134 @@
+const { Amount, pkg: hsdPkg } = require('hsd');
+const assert = require('bsert');
+
+const { isChainSupported } = require('./chain');
+
+export const CURRENCY_UNITS = {
+  bitcoin: {
+    currency: 'bitcoin',
+    unit: 'btc',
+    milli: 'mbtc',
+    micro: 'bit',
+    base: 'satoshi',
+  },
+  handshake: {
+    currency: hsdPkg.currency,
+    unit: hsdPkg.unit,
+    milli: `m${hsdPkg.unit}`,
+    micro: `u${hsdPkg.unit}`,
+    base: hsdPkg.base,
+  },
+};
+
+export const CURRENCY_TYPES = {
+  currency: 'currency',
+  unit: 'unit',
+  milli: 'milli',
+  micro: 'micro',
+  base: 'base',
+};
+// bitcoincash units currently same as bitcoin
+CURRENCY_UNITS.bitcoincash = { ...CURRENCY_UNITS.bitcoin };
+
+class Currency extends Amount {
+  /**
+   * Create an object for getting currency units and amounts
+   * @constructor
+   * @param {string} chain - one of supported chain types
+   * see { chain.CHAINS }
```

Comment:
is this `bpanel-utils/chain.CHAINS`?

Reply:
yep!

Reply:
I was more doing it as a jsdocs indication. Even so this isn't quite right. so probably should be `{@link module: chain.CHAINS}`

Reply:
Let me know if you think that's appropriate/good enough or if still preferable to do `bpanel-utils/chain.CHAINS` and I'll update. Agree that this is too ambiguous though.

Reply:
I think thats good 👍